src/platform/imx8ulp/include/platform/lib/memory.h

```diff
@@ -0,0 +1,220 @@
+
+#define SRAM_INBOX_BASE		(SRAM_OUTBOX_BASE + SRAM_OUTBOX_SIZE)
+#define SRAM_INBOX_SIZE		0x1000
+#define SRAM_INBOX_OFFSET	SRAM_OUTBOX_SIZE
+
+#define SRAM_DEBUG_BASE		(SRAM_INBOX_BASE + SRAM_INBOX_SIZE)
+#define SRAM_DEBUG_SIZE		0x2800
+#define SRAM_DEBUG_OFFSET	(SRAM_INBOX_OFFSET + SRAM_INBOX_SIZE)
+
+#define SRAM_EXCEPT_BASE	(SRAM_DEBUG_BASE + SRAM_DEBUG_SIZE)
+#define SRAM_EXCEPT_SIZE	0x800
+#define SRAM_EXCEPT_OFFSET	(SRAM_DEBUG_OFFSET + SRAM_DEBUG_SIZE)
+
+#define SRAM_STREAM_BASE	(SRAM_EXCEPT_BASE + SRAM_EXCEPT_SIZE)
+#define SRAM_STREAM_SIZE	0x1000
+#define SRAM_STREAM_OFFSET	(SRAM_EXCEPT_OFFSET + SRAM_EXCEPT_SIZE)
+
+#define SRAM_TRACE_BASE		(SRAM_STREAM_BASE + SRAM_STREAM_SIZE)
+#define SRAM_TRACE_SIZE		0x1000
+#define SRAM_TRACE_OFFSET (SRAM_STREAM_OFFSET + SRAM_STREAM_SIZE)
+
+#define SOF_MAILBOX_SIZE	(SRAM_INBOX_SIZE + SRAM_OUTBOX_SIZE \
+				+ SRAM_DEBUG_SIZE + SRAM_EXCEPT_SIZE \
+				+ SRAM_STREAM_SIZE + SRAM_TRACE_SIZE)
+
+/* Heap section sizes for module pool */
+#define HEAP_RT_COUNT8		0
+#define HEAP_RT_COUNT16		48
+#define HEAP_RT_COUNT32		48
+#define HEAP_RT_COUNT64		32
+#define HEAP_RT_COUNT128	32
+#define HEAP_RT_COUNT256	32
+#define HEAP_RT_COUNT512	4
+#define HEAP_RT_COUNT1024	4
+#define HEAP_RT_COUNT2048	4
+
+/* Heap section sizes for system runtime heap */
+#define HEAP_SYS_RT_COUNT64	128
+#define HEAP_SYS_RT_COUNT512	16
+#define HEAP_SYS_RT_COUNT1024	8
+
+/* Heap configuration */
+
+#define HEAP_SYSTEM_BASE	(SDRAM1_BASE + SOF_MAILBOX_SIZE)
+#define HEAP_SYSTEM_SIZE	0xe000
+
+#define HEAP_SYSTEM_0_BASE	HEAP_SYSTEM_BASE
+
+#define HEAP_SYS_RUNTIME_BASE	(HEAP_SYSTEM_BASE + HEAP_SYSTEM_SIZE)
+#define HEAP_SYS_RUNTIME_SIZE \
+	(HEAP_SYS_RT_COUNT64 * 64 + HEAP_SYS_RT_COUNT512 * 512 + \
+	HEAP_SYS_RT_COUNT1024 * 1024)
+
+#define HEAP_RUNTIME_BASE	(HEAP_SYS_RUNTIME_BASE + HEAP_SYS_RUNTIME_SIZE)
+#define HEAP_RUNTIME_SIZE \
+	(HEAP_RT_COUNT8 * 8 + HEAP_RT_COUNT16 * 16 + \
+	HEAP_RT_COUNT32 * 32 + HEAP_RT_COUNT64 * 64 + \
+	HEAP_RT_COUNT128 * 128 + HEAP_RT_COUNT256 * 256 + \
+	HEAP_RT_COUNT512 * 512 + HEAP_RT_COUNT1024 * 1024 + \
+	HEAP_RT_COUNT2048 * 2048)
+
+#define HEAP_BUFFER_BASE	(HEAP_RUNTIME_BASE + HEAP_RUNTIME_SIZE)
+#define HEAP_BUFFER_SIZE \
+	(SDRAM1_SIZE - SOF_MAILBOX_SIZE - HEAP_RUNTIME_SIZE - SOF_STACK_TOTAL_SIZE -\
+	HEAP_SYS_RUNTIME_SIZE - HEAP_SYSTEM_SIZE)
+
+#define HEAP_BUFFER_BLOCK_SIZE		0x100
+#define HEAP_BUFFER_COUNT	(HEAP_BUFFER_SIZE / HEAP_BUFFER_BLOCK_SIZE)
+
+#define PLATFORM_HEAP_SYSTEM		1 /* one per core */
+#define PLATFORM_HEAP_SYSTEM_RUNTIME	1 /* one per core */
+#define PLATFORM_HEAP_RUNTIME		1
+#define PLATFORM_HEAP_BUFFER		1
+
+/* Stack configuration */
+#define SOF_STACK_SIZE		0x1000
+#define SOF_STACK_TOTAL_SIZE	SOF_STACK_SIZE
+#define SOF_STACK_BASE		(SDRAM1_BASE + SDRAM1_SIZE)
+#define SOF_STACK_END		(SOF_STACK_BASE - SOF_STACK_TOTAL_SIZE)
+
+/* Vector and literal sizes - not in core-isa.h */
+#define SOF_MEM_VECT_LIT_SIZE		0x4
+#define SOF_MEM_VECT_TEXT_SIZE		0x1c
```

Comment:
Where did you take these values from?

Reply:
This part is just a copy from the imx8m platform.